bin/i18n/translation_status/update_translation_status.rb

```diff
@@ -4,20 +4,103 @@
   query = <<~SQL.squish
+    DELETE FROM analysis.i18n_string_translation_status
+    WHERE string_key IN (
+      #{unique_string_key_query}
+    )
+  SQL
+  redshift_client.exec(query)
+end
+
+# Retrieves all the unique string_key's we have logged, checks if they are translated or not in each language, and then
+# records that info to the analysis.i18n_string_translation_status table.
+# @param [Integer] day_count The number of days in the past to look for unique string_key's
+def insert_translation_status(redshift_client, day_count)
+  # Retrieve all the unique string_key's we have logged in the analysis.i18n_string_tracking_events.
+  unique_string_key_query = unique_string_key_sql_query(day_count)
+  string_keys = redshift_client.exec(unique_string_key_query).map {|row| row['string_key']}
+
+  # Used to determine if a given string_key is translated or not.
+  translation_service = TranslationService.new
+  # All the locale codes we support i.e. en_US, es_MX, de_DE, etc.
+  locales = Languages.get_locale.map {|lang| lang[:locale_s]}
+  # Record what time we checked the translation status for these strings. Use the same time for all the entries
+  # so we can tell which batch they were all checked in.
+  # Timestamp format matches the one supported by Redshift.
+  current_time = Time.now.utc.strftime("%Y-%m-%d %H:%M:%S")
+  # Loop through each locale code our platform supports, i.e. en_US, es_MX, de_DE, etc. We want to check the translation
+  # status in each language for every string_key.
+  locales.each_with_index do |locale, i|
+    puts "Analyzing translation status of #{locale} (#{i + 1}/#{locales.size})"
+    translation_statuses = []
+    # Record the translation status for every string_key, i.e. "data.script.name.express-2020.title"
+    string_keys.each do |string_key|
+      # Maybe there is bad data in the database and we recorded an empty string
+      next if string_key.blank?
+      translated = translation_service.translated?(locale, string_key)
+      translation_statuses << {
+        string_key: string_key,
+        locale: locale,
+        has_translation: translated,
+        checked_at: current_time
+      }
+    end
+    # Create the rows to be inserted in the analysis.i18n_string_translation_status Redshift table
+    # https://docs.aws.amazon.com/redshift/latest/dg/c_Examples_of_INSERT_30.html
+    values = translation_statuses.map do |status|
+      # Escape special SQL characters to protect against SQL injection attacks.
+      ActiveRecord::Base.sanitize_sql(
+        ["('%s', '%s', '%s', '%s')",
+         status[:string_key],
+         status[:locale],
+         status[:has_translation],
+         status[:checked_at]]
+      )
+    end.join(',')
+    # Run a Redshift query to insert the translation statuses into the analysis.i18n_string_translation_status table.
+    query = <<~SQL.squish
+      INSERT INTO analysis.i18n_string_translation_status (string_key, locale, has_translation, checked_at)
```

Comment:
Might be useful to pull these table names out into constants

Reply:
done